### PR TITLE
Create a library for maestro nodejs support

### PR DESCRIPTION
**Pull request by galosa**
This library will contain all necessary component for a service to be able to orchestrate with Maestro.
Resolve [INF-438, INF-439]

### CODE REVIEW

File: JenkinsFile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env groovy
+
+nodeJsLibraryBuild {
+    project = "maestro-js-support"
```
Comment (dshush) [inline JenkinsFile]: I saw that Erik added this line also: (in the backend common)

testCommand = 'npm run test -- --reporter=xunit --reporter-options output=./reports.xml'

File: library/index.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const internalRoutes = require('./routes/internal');
```
Comment (galosa) [inline library/index.js]: No, i will remove it.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "maestro-js-support",
+  "version": "4.6.0",
+  "description": "Maestro NodeJs support library",
+  "main": "library/index.js",
+  "scripts": {
```
Comment (galosa) [inline package.json]: I have `lint` 
Do you also mean `lint-loud`? and `lint-fix`?

File: lib/index.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const internalRoutes = require('./routes/internal');
+const PhaseSupport = require('./scenario/phase.support').default;
+const MongoStore = require('./scenario/store/mongo.store').default;
+
+
+export default function (serviceName, app, dbConnection, logger, organizationProvider) {
+  const store = MongoStore(dbConnection);
+  return {
+    // TODO: change organizationProvide only to expose findOne method and not the all interface
```
Comment (iravid) [inline lib/index.js]: This still isn't done right?

File: lib/routes/internal.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+const ScenarioSupport = require('../scenario/scenario.support').default;
+const HttpErrorHandler = require('../scenario/error.handling').HttpErrorHandler;
+const ScenarioHandler = require('../scenario/handlers/scenario').default;
+const UUID = require('uuid');
+const Schema = require('../scenario/scenarioRequestSchema');
+
+// TODO: change to export default
```
Comment (iravid) [inline lib/routes/internal.js]: Going to do this?

Comment (galosa) [inline lib/routes/internal.js]: Yes

File: lib/index.js (hunk 4 above)
Comment (galosa) [inline lib/index.js]: Doing it, it will be implemented in the services and in the lib.

**Comment by galosa** [comment]
@iravid @dshush please re-approve so I can merge

File: lib/scenario/scenario.support.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      return Promise.reject(new Error(ERRORS.ScenarioInterrupted));
+    }
+    return Promise.resolve();
+  }
+
+  function isOrganizationExist(orgName) {
+    if (!orgName) {
+      return Promise.reject(new Error(ERRORS.OrgNotExist));
+    }
+    return organizationProvider.findOne({ name: orgName }).exec().then((org) => {
```
Comment (iravid) [inline lib/scenario/scenario.support.js]: Didn't we say that the `organizationProvider` should just be a function that returns the org?

**Pull request merged**